### PR TITLE
fix(query): fix cast deterministic error

### LINKED ISSUE #7682: bug: `cast(rand() as string)` returns the same result
**Summary**

```
mysql> select rand() from numbers(10);
+----------------------+
| rand()               |
+----------------------+
| 0.016333804973599486 |
|  0.49383331056312574 |
|   0.9841902151336861 |
| 0.043982804663117325 |
|   0.2259206402471785 |
|  0.00997215327383083 |
|   0.7977313791817102 |
|   0.7871726221864119 |
|   0.9605240485760037 |
|  0.30119168210376523 |
+----------------------+
10 rows in set (0.01 sec)
Read 10 rows, 80.00 B in 0.002 sec., 4.35 thousand rows/sec., 34.02 KiB/sec.

mysql> select cast(rand() as string) from numbers(10);
+------------------------+
| cast(rand() as string) |
+------------------------+
| 0.9355074606222086     |
| 0.9355074606222086     |
| 0.9355074606222086     |
| 0.9355074606222086     |
| 0.9355074606222086     |
| 0.9355074606222086     |
| 0.9355074606222086     |
| 0.9355074606222086     |
| 0.9355074606222086     |
| 0.9355074606222086     |
+------------------------+
```

### PR DESCRIPTION
**Pull request by b41sh**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

fix cast deterministic error

Fixes #7682

### CODE REVIEW

File: src/query/service/src/sql/planner/plans/scalar.rs [hunk 1]
```diff
@@ -511,7 +511,7 @@ impl ScalarExpr for CastExpr {
     }
 
     fn is_deterministic(&self) -> bool {
-        true
+        false
```
Comment (sundy-li) [inline src/query/service/src/sql/planner/plans/scalar.rs:514]: `Cast` should be deterministic.

`random` is not deterministic, so `cast(random())` is not deterministic.  We should find the root reason.

Comment (sundy-li) [reply src/query/service/src/sql/planner/plans/scalar.rs:514]: should be `self.argument.is_deterministic()`

Comment (b41sh) [reply src/query/service/src/sql/planner/plans/scalar.rs:514]: ok

**Review by sundy-li — APPROVED** [review]

**Review by Xuanwo — APPROVED** [review]

File: src/query/service/src/sql/planner/plans/scalar.rs [hunk 2]
```diff
@@ -511,7 +511,7 @@ impl ScalarExpr for CastExpr {
     }
 
     fn is_deterministic(&self) -> bool {
-        true
+        self.argument.is_deterministic()
```
Comment (Xuanwo) [inline src/query/service/src/sql/planner/plans/scalar.rs:514]: Please add a sqllogic test for this.

Comment (Xuanwo) [reply src/query/service/src/sql/planner/plans/scalar.rs:514]: We can compare the result of two `cast(rand() as string)`

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**